source/fundamentals/linq.txt

```diff
@@ -0,0 +1,541 @@
+
+The preceding example returns the following document:
+
+.. code-block:: json
+
+    // Results Truncated
+
+    { "_id" : ObjectId(...), "name" : "The Movable Feast", "restaurant_id" : "40361606", "cuisine" : "American", "address" : {...}, "borough" : "Brooklyn", "grades" : [...] }
+
+$limit
+~~~~~~
+
+The ``$limit`` aggregation stage limits the number of documents returned by the
+query. The following example shows how to generate a ``$limit`` stage using LINQ:
+
+.. code-block:: csharp
+   :emphasize-lines: 4
+
+   var query = queryableCollection
+       .Where(r => r.Cuisine == "Italian")
+       .Select(r => new {r.Name, r.Cuisine})
+       .Take(5);
+
+The preceding example returns the following documents:
+
+.. code-block:: json
+
+    { "name" : "Philadelhia Grille Express", "cuisine" : "Italian" }
+    { "name" : "Isle Of Capri Resturant", "cuisine" : "Italian" }
+    { "name" : "Marchis Restaurant", "cuisine" : "Italian" }
+    { "name" : "Crystal Room", "cuisine" : "Italian" }
+    { "name" : "Forlinis Restaurant", "cuisine" : "Italian" }
+
+$skip
+~~~~~
+
+The ``$skip`` aggregation stage skips over a specified number of documents returned
+by a query, then returns the rest of the results. The following example shows how to generate
+a ``$skip`` stage using LINQ:
+
+.. code-block:: csharp
+   :emphasize-lines: 4
+
+   var query = queryableCollection
+       .Where(r => r.Cuisine == "Italian")
+       .Select(r => new {r.Name, r.Cuisine})
+       .Skip(2);
+
+The preceding example skips the first two restaurants that match the criteria, and
+returns the rest as follows:
+
+.. code-block:: json
+
+   // Results Truncated
+
+   { "name" : "Marchis Restaurant", "cuisine" : "Italian" }
+   { "name" : "Crystal Room", "cuisine" : "Italian" }
+   { "name" : "Forlinis Restaurant", "cuisine" : "Italian" }
+   ...
+
+$unwind
+~~~~~~~
+
+The ``$unwind`` aggregation stage deconstructs a specified array field and returns
+a document for each element in that array.
+
+Select the :guilabel:`Query Syntax` or :guilabel:`Method Syntax` tab to see how 
+to generate an ``$unwind`` stage using LINQ:
+
+.. tabs::
+
+   .. tab:: Query Syntax
+      :tabid: query-syntax
+
+      .. code-block:: csharp
+         :emphasize-lines: 3
+
+         var query = from r in queryableCollection
+             where r.Name == "The Movable Feast"
+             from grade in r.Grades
+             select grade;
+
+   .. tab:: Method Syntax
+      :tabid: method-syntax
+
+      .. code-block:: csharp
+         :emphasize-lines: 3
+
+         var query = queryableCollection
+             .Where(r => r.Name == "The Movable Feast")
+             .SelectMany(r => r.Grades);
+
+The query in the preceding example finds the document where the ``Name`` field 
+has the value "The Movable Feast." Then, for each element in this document's 
+``Grades`` array, the query returns a new document. The results are as follows:
+
+.. code-block:: json
+
```

Comment:
I did execute the query and it is correct, however the output is from iterating over the returned documents. I don't think it was very clear that that's what was happening so I reworded the paragraph to state that the result contains these documents, rather than saying that this is what's returned.

Reply:
Where is the `Grade` class declared? This example is incomplete if we can't tell what a `Grade` looks like.

Unless you want to force readers to "infer" what a `Grade` class *might* look like given the resulting output.

And again, they'd have to know that you are assuming non-default conventions, so you'd have to point that out again here and everywhere if you don't want readers to miss that important detail.